core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -328,7 +315,7 @@ private Object primaryReturnHandler(InvocationContext ctx, VisitableCommand visi
       }));
    }
 
-   private RpcOptions determineRpcOptionsForBackupReplication(RpcManager rpc, boolean isSync, List<Address> recipients) {
+   private RpcOptions determineRpcOptionsForBackupReplication(RpcManager rpc, boolean isSync, Collection<Address> recipients) {
```

Comment:
So  does rpc.getDefaultRpcOptions(true);
And you can cache rpcs options for `S_I_L`, too.